rustls/src/msgs/handshake_test.rs

```diff
@@ -358,28 +358,34 @@ fn get_sample_clienthellopayload() -> ClientHelloPayload {
         client_version: ProtocolVersion::TLSv1_2,
         random: Random::from([0; 32]),
         session_id: SessionID::empty(),
-        cipher_suites: vec![CipherSuite::TLS_NULL_WITH_NULL_NULL],
-        compression_methods: vec![Compression::Null],
+        cipher_suites: vec![CipherSuite::TLS_NULL_WITH_NULL_NULL].into(),
```

Comment:
It seems wrong we're using `vec!` here and elsewhere. I would expect `TlsVec::from(&[CipherSuite::TLS_NULL_WITH_NULL_NULL])` or so. Or otherwise a `tls_vec!`.